backend/common/src/main/java/ai/verta/modeldb/common/authservice/AuthServiceChannel.java

```diff
@@ -56,10 +56,10 @@ public AuthServiceChannel(
   }
 
   private Metadata getMetadataHeaders() {
-    int backgroundUtilsCount = CommonUtils.getRegisteredBackgroundUtilsCount();
-    LOGGER.trace("Header attaching with stub : backgroundUtilsCount : {}", backgroundUtilsCount);
+    boolean metadataNotExists = metadataInfo == null || metadataInfo.get() == null;
+    LOGGER.trace("Header info exists: {}", metadataNotExists);
     Metadata requestHeaders;
-    if (backgroundUtilsCount > 0 && (metadataInfo == null || metadataInfo.get() == null)) {
+    if (metadataNotExists) {
```

Comment:
I'd rather get rid of this altogether. It's an implicit thing whether to use a service account or not, which can easily be done by mistake and causes all sort of errors. I'd like the use of service account be a deliberate choice instead of the default to avoid these bugs.

Reply:
we have already make a service account choice deliberate from here https://github.com/VertaAI/modeldb/blob/b224fb60835c0e83440d3583eab15700a02e8269/backend/common/src/main/java/ai/verta/modeldb/common/authservice/RoleServiceUtils.java#L123 and https://github.com/VertaAI/modeldb/blob/b224fb60835c0e83440d3583eab15700a02e8269/backend/common/src/main/java/ai/verta/modeldb/common/authservice/RoleServiceUtils.java#L128 so whatever used auth channel they know the user preference  and they will initialize service based on the same.

Reply:
You are assuming we're always implementing the code correctly when defining those methods. You should not make this assumption. If we pass `null` in the argument by mistake (which is _very_ easy because the variable itself can have value `null` so it's not even a typo in the code), then we'll use the superuser credentials. That's not ok. The default behavior must be safety: do not assume that a missing value means the call should be made with a super user; force the caller to say "yes, I want to do this as a super user".

Reply:
Updated PR with the your suggestion. Now all auth channel logic is implemented and managed by UAC.java class. The Service account status is now separated.